openhtf/__init__.py

```diff
@@ -161,14 +162,27 @@ def plug_type_map(self):
     return plug_type_map
 
 
-def SetupFramework():
-  """Sets up various bits of the framework. Only needs to be called once."""
-  try:
-    sys.argv = FLAGS(sys.argv)  # parse flags
-  except gflags.FlagsError, e:  # pylint: disable=invalid-name
-    print '%s\nUsage: %s ARGS\n%s' % (e, sys.argv[0], FLAGS)
-    sys.exit(1)
+def CreateArgParser():
+  """Creates an argparse.ArgumentParser for parsing command line flags."""
+  parser = argparse.ArgumentParser('OpenHTF-based testing')
+  # openhtf.io.user_input
+  # Option 1:
+  parser.add_argument('--prompt_timeout_s', type=int,
+                      help='User prompt timeout in seconds.')
+  # Option 2:
+  parser = argparse.ArgumentParser('OpenHTF-based testing', parents=[
+      user_input.DEFAULT_TIMEOUT_S_PARSER])
+
+
+  return parser
 
+
+@functions.RunOnce
+def SetupFramework(arguments=None):
+  """Sets up various bits of the framework. Only needs to be called once."""
+  parser = arguments or CreateArgParser().parse_args()
+  # Option 3:
```

Comment:
This isn't the only place this comes up, so the fourth option is pretty untenable, unfortunately. CodeInfo does it too, and worse